### PR TITLE
feat: docs breadcrumbs

### PR DESCRIPTION
**Pull request by fsansalvadore**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Adds breadcrumbs to docs guides pages.

## What is the current behavior?

No breadcrumbs.

## What is the new behavior?

Screenshot 2024-07-11 at 15 51 06

Screenshot 2024-07-11 at 15 50 58

### CODE REVIEW

**Comment by MildTomato** [comment]
Think we need more varation of color on these breadcrumbs. 
See the Design System site for what I mean.

Screenshot 2024-07-11 at 3 53 42 PM

File: apps/docs/components/Breadcrumbs.tsx
Comment (fsansalvadore) [inline apps/docs/components/Breadcrumbs.tsx:110]: @MildTomato don't you think this draws too much focus on the last breadcrumb itself?
(which also highlights the redundancy a bit too much imo)

Screenshot 2024-07-11 at 16 46 20

File: apps/docs/features/docs/GuidesMdx.template.tsx
Comment (charislam) [inline apps/docs/features/docs/GuidesMdx.template.tsx:171]:
```suggestion
            // 3rem accounts for 2rem of top padding + 1rem of extra breathing room
```

File: apps/docs/features/docs/GuidesMdx.template.tsx
Comment (charislam) [inline apps/docs/features/docs/GuidesMdx.template.tsx:124]: Do you think the spacing on the bottom is a little tight? Your final call on this (you're the expert 😄)

image

File: packages/ui/src/components/shadcn/ui/breadcrumb.tsx
Comment (charislam) [inline packages/ui/src/components/shadcn/ui/breadcrumb.tsx:101]: Hmm I think the `sr-only` within the `aria-hidden` isn't going to show up for screen readers, maybe make it a sibling element or put the label on the other span and put the `aria-hidden` on the SVG?

**Review by charislam — APPROVED** [review]
Approving to unblock from the docs side, came across a few nits:

On the Main Auth page, a redundant breadcrumb:

image

On mobile, it's not entirely clear which breadcrumbs are clickable and which aren't, so I got a bit frustrated trying to click unclickable ones before realizing the problem 🤦🏼‍♀️ 

Love the mobile drawer though! ✨ 

Checked:
- Checked a couple pages in docs preview (desktop and mobile simulator) + tried the "show more" option
- Opened other previews

File: apps/docs/components/Breadcrumbs.tsx
Comment (MildTomato) [reply apps/docs/components/Breadcrumbs.tsx:110]: I've seen it both ways. Although the trend I see is last item has most dominance.
Screenshot 2024-07-11 at 7 07 13 PM

Comment (fsansalvadore) [reply apps/docs/components/Breadcrumbs.tsx:110]: I think it makes more sense if it's not close to the title of the page.
But lets go for dominance if you think that's the way 👍

File: apps/docs/features/docs/GuidesMdx.template.tsx
Comment (fsansalvadore) [reply apps/docs/features/docs/GuidesMdx.template.tsx:124]: you're right, mb-2 is better 😁

**Comment by fsansalvadore** [comment]
@charislam I made the links underlined on mobile, if they are clickable.

> On the Main Auth page, a redundant breadcrumb:

This is fixed now.

File: apps/docs/components/Breadcrumbs.tsx
Comment (fsansalvadore) [reply apps/docs/components/Breadcrumbs.tsx]: but using `foreground-light` seems like a good compromise 👍

Comment (fsansalvadore) [reply apps/docs/components/Breadcrumbs.tsx]: Screenshot 2024-07-11 at 19 54 12

File: packages/ui/src/components/shadcn/ui/breadcrumb.tsx
Comment (charislam) [reply packages/ui/src/components/shadcn/ui/breadcrumb.tsx:96]: yeah I think that's fine

File: apps/docs/components/Breadcrumbs.tsx
Comment (fsansalvadore) [inline apps/docs/components/Breadcrumbs.tsx:42]: @charislam @MildTomato I opted to omit the last breadcrumb on mobile and let intend that the title is the conclusion of the breadcrumb to save real estate and avoid too many multiple line breadcrumbs situations. Wdyt?

Desktop:
Screenshot 2024-07-12 at 19 19 08

Mobile:
Screenshot 2024-07-12 at 19 19 58

Comment (fsansalvadore) [reply apps/docs/components/Breadcrumbs.tsx:42]: Otherwise it would be like this.
I think we should optimise for mobile, this is too crowded.

Screenshot 2024-07-12 at 19 22 34

Comment (MildTomato) [reply apps/docs/components/Breadcrumbs.tsx:42]: yea sure 👍

**Pull request merged**
